BootloaderCorePkg/Library/MpInitLib/Ia32/MpFuncs.nasm

```diff
@@ -23,23 +23,36 @@ SECTION .text
 ;-------------------------------------------------------------------------------------
 ;RendezvousFunnelProc  procedure follows. All APs execute their procedure. This
 ;procedure does NOT serialize the AP processors. So this procedure cannot be the
-;vector for a boradcast IPI. 
-; 
+;vector for a boradcast IPI.
+;
 ;It must be noted that APs arrive here very raw...ie: real mode, no stack.
 ;-------------------------------------------------------------------------------------
 SPIN_LOCK_RELEASED   EQU   0
 SPIN_LOCK_ACQUIRED   EQU   1
 
- 
+
 global ASM_PFX(RendezvousFunnelProc)
 ASM_PFX(RendezvousFunnelProc):
 RendezvousFunnelProcStart:
 ; At this point CS = 0x(vv00) and ip= 0x0.
-  
+  db 0fh,  20h, 0d0h            ; mov        eax, cr2
```

Comment:
Why not use the real instructions here?

Reply:
At this point, the CPU is still in real mode.  But the assembler will try to create 32bit instruction which might cause compatibility issue.  That is why the opcode is used directly until the CPU is fully in 32bit flat mode.

Reply:
NASM supports assembling/emitting 16-bit instructions if you specify "bits 16" just before the instructions (then back to 32b with a "bits 32" line). I think this is fine for now but we might want to change it to make it more maintainable down the road.

Reply:
Yes, agree.   I can try to explore a little bit on this to see if it can be changed to use mixed ASM code.  If so, we can do it in a new PR.

Reply:
BTW, submitted a new PR #37 to convert opcode into instructions.

Reply:
Awesome, thx @mauricema!